packages/arb-bridge-eth/contracts/inbox/GlobalFTWallet.sol

```diff
@@ -69,13 +86,74 @@ contract GlobalFTWallet {
         return wallet.ftList[index - 1].balance;
     }
 
+    function isPairedContract(address _tokenContract, address _chain)
+        public
+        view
+        returns (bool)
+    {
+        return
+            pairedContracts[_tokenContract].connectedRollups[_chain] ==
+            PairingStatus.Paired;
+    }
+
+    function requestPairing(address _tokenContract, address _chain) internal {
+        PairedContract storage pairedContract = pairedContracts[_tokenContract];
+        require(
+            pairedContract.connectedRollups[_chain] == PairingStatus.Unpaired,
+            "must be unpaired"
+        );
+        if (!pairedContract.paired) {
+            // This is the first time pairing with a chain
+            pairedContract.paired = true;
+
+            // Burn existing balance since we will switch over to minting on withdrawal
+            IPairedErc20 tokenContract = IPairedErc20(_tokenContract);
```

Comment:
This also seems to assume that the L2 part of the pairing cannot fail.  It seems safer to burn these when the pairing is confirmed.